doc/cug.tex

```diff
@@ -3789,30 +3789,124 @@ \subsubsection{Trigger Types}
+on an ftp server, and then notify a suite containing a task to retrieve the
+new data for processing. This is an alternative to having long-running tasks
+that poll for external events.
+
+Note that cylc does not directly support ``filesystem events'' because systems
+such as \lstinline=inotify= on Linux are not portable. However, external
+watcher processes can of course use inotify to detect filesystem events, if
+that is appropriate, before triggering the suite.
+
+External triggers are not normally needed in date-time cycling suites driven
+by real time data that comes in at regular intervals.  In these cases a data
+retrieval task can be clock-triggered to submit at the expected data arrival
+time, so little time if any is wasted in polling.  But in non-cycling or
+integer cycling suites (which can be used for arbitrarily timed repeating
+workflows such as for satellite data processing) external triggers may be
+preferred.
+
+The external triggering process must call \lstinline=cylc ext-trigger= with the
+name of the target suite, the message that identifies this type of event to the
+suite, and an ID that distinguishes this particular event instance from others
+(the name of the target task or its current cycle point is not required).  The
+event ID is just an arbitary string to cylc, but it typically identifies the
+filename(s) of the latest dataset in some way.  When the suite daemon receives
+the external event notification it will trigger the next instance of any task
+waiting on that trigger (whatever its cycle point) and then broadcast
+(see~\ref{cylc-broadcast}) the event ID to the cycle point of the triggered
+task as \lstinline=$CYLC_EXT_TRIGGER_ID=.  Downstream tasks with the same cycle
+point therefore know the new event ID too and can use it, if they need to, to
+identify the same new dataset. In this way a whole workflow can be associated
+with each new dataset, and multiple datasets can be processed in parallel if
+they happen to arrive in quick succession.  
+
+An externally-triggered task must register the event it waits on in the suite
+scheduling section:
+\lstset{language=suiterc}
+\begin{lstlisting}
+# suite "sat-proc"
+[scheduling]
+    cycling mode = integer
+    initial cycle point = 1
+    [[special tasks]]
+        external-triggered = get-data("new sat X data avail")
+    [[dependencies]]
+        [[[P1]]]
+            graph = get-data => conv-data => products
+\end{lstlisting}
+
+Then, each time a new dataset arrives the external detection system should
+notify the suite like this:
+\lstset{language=transcript}
+\begin{lstlisting}
+shell$ cylc ext-trigger sat-proc "new sat X data avail" passX12334a
+\end{lstlisting}
+where ``sat-proc'' is the suite name and ``passX12334a'' is the ID string for
+the new event.
+
+Note that only one task in a suite can trigger off a given external message.
+This is no restriction however because other tasks can trigger off the
+externally triggered task.
+
```

Comment:
Presumably can also only issue external triggers to suites for which you have the passphrase? If so, worth mentioning as people might think to use this mechanism to trigger off other people's systems/suites etc.